server/src/controllers/surveyController.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as express from 'express';
+import SurveyDBHandler from '../databases/handlers/surveyDBHandler';
+import { SurveyResultRequestModel } from '../models/surveyModel';
+
+export const storeSurveyResult = (surveyDBHandler: SurveyDBHandler) => async (
+  req: express.Request,
+  res: express.Response,
+  next: express.NextFunction
+): Promise<any> => {
+  try {
+    const { body: requestData } = req;
+    const { sessionId, callId, acsUserId, response } = requestData;
+
+    // Validation.
+    // If any one of field does not exists in the payload the validation failed.
+    if (!(sessionId && callId && acsUserId && response)) {
+      const errors: string[] = [];
+
+      if (!sessionId) errors.push('sessionId is missing');
+      if (!callId) errors.push('callId is missing');
+      if (!acsUserId) errors.push('acsUserId is missing');
+      if (!response) errors.push('response is missing');
+
+      return res.status(400).send({ errors });
+    }
+
+    const inputData: SurveyResultRequestModel = {
+      ...requestData,
+      createdOn: new Date()
+    };
+
+    await surveyDBHandler.init();
```

Comment:
Shouldn't `init` on every controller request. Do this once on the app startup instead.

Reply:
I tried to `init` in `app.ts` but `init` is asynchronous function that I wasn't able to run it on `app.ts`. Do you know the way I can run asynchronous function in `app.ts` somehow?

Reply:
Because of the above reason, inside both [`createDatabase`](https://github.com/Azure-Samples/communication-services-virtual-visits-js/pull/96/files#diff-f8bfe563ab4c43a502d9afa610dbdb917b21a0f688128c45fbfbe2825aa79427:~:text=if%20(this._database%20%3D%3D%3D%20undefined)%20%7B) and [`createContainer`](if (this._container === undefined) {) have if statement to not invoke `createIfNotExsists` method at all.

```javascript
if (this._database === undefined) {
...
}

if (this._container === undefined) {
...
}
```